pkg/network/network_filter.go

```diff
@@ -164,12 +164,12 @@ func IsExcludedConnection(scf []*ConnectionFilter, dcf []*ConnectionFilter, conn
 	buf := util.IPBufferPool.Get().(*[]byte)
 	defer util.IPBufferPool.Put(buf)
 
-	if len(scf) > 0 && conn.Source != nil {
+	if len(scf) > 0 {
```

Comment:
What do you think about updating the filter logic to use the `netaddr` package, specifically the `netaddr.IPPrefix` type? We could get rid of the `IPBufferPool` usage here then.